src/_pytest/runner.py

```diff
@@ -168,6 +168,8 @@ def pytest_runtest_call(item: Item) -> None:
         del sys.last_type
         del sys.last_value
         del sys.last_traceback
+        if sys.version_info >= (3, 12, 0):
+            del sys.last_exc
```

Comment:
Turns it the type stub for it is just missing from typeshed (the project which provides type annotations for the standard library). I sent [a fix](https://github.com/python/typeshed/pull/11378) for it, but until it's integrated (in a future mypy version), let's add type-ignores for it, like this:

```suggestion
            del sys.last_exc  # type: ignore[attr-defined]
```